kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DomainMetadataSuite.scala

```diff
@@ -0,0 +1,472 @@
+  }
+
+  def setDomainMetadataSupport(engine: Engine, tablePath: String): Unit = {
+    val protocol = new Protocol(
+      1, // minReaderVersion
+      7, // minWriterVersion
+      Collections.emptyList(), // readerFeatures
+      Seq("domainMetadata").asJava // writerFeatures
+    )
+
+    val protocolAction = SingleAction.createProtocolSingleAction(protocol.toRow)
+    val txn = createTxn(engine, tablePath, isNewTable = false, testSchema, Seq.empty)
+    txn.commit(engine, inMemoryIterable(toCloseableIterator(Seq(protocolAction).asJava.iterator())))
+  }
+
+  def createTableWithDomainMetadataSupported(engine: Engine, tablePath: String): Unit = {
+    // Create an empty table
+    createTxn(engine, tablePath, isNewTable = true, testSchema, Seq.empty)
+      .commit(engine, emptyIterable())
+
+    // Set writer version and writer feature to support domain metadata
+    setDomainMetadataSupport(engine, tablePath)
+  }
+
+  private def validateDomainMetadataConflictResolution(
+      engine: Engine,
+      tablePath: String,
+      currentTxn1DomainMetadatas: Seq[DomainMetadata],
+      winningTxn2DomainMetadatas: Seq[DomainMetadata],
+      winningTxn3DomainMetadatas: Seq[DomainMetadata],
+      expectedConflict: Boolean): Unit = {
+
+    // Create table with domain metadata support
+    createTableWithDomainMetadataSupported(engine, tablePath)
+    val table = Table.forPath(engine, tablePath)
+
+    /**
+     * Txn1: i.e. the current transaction that comes later than winning transactions.
+     * Txn2: i.e. the winning transaction that was committed first.
+     * Txn3: i.e. the winning transaction that was committed secondly.
+     *
+     * Note tx is the timestamp.
+     *
+     * t1 ------------------------ Txn1 starts.
+     * t2 ------- Txn2 starts.
+     * t3 ------- Txn2 commits.
+     * t4 ------- Txn3 starts.
+     * t5 ------- Txn3 commits.
+     * t6 ------------------------ Txn1 commits (SUCCESS or FAIL).
+     */
+    try {
+      val txn1 = createTxn(engine, tablePath, isNewTable = false, testSchema, Seq.empty)
+
+      val txn2 = createTxn(engine, tablePath, isNewTable = false, testSchema, Seq.empty)
+      txn2.commit(engine, generateDMActionsForCommit(winningTxn2DomainMetadatas))
+
+      val txn3 = createTxn(engine, tablePath, isNewTable = false, testSchema, Seq.empty)
+      txn3.commit(engine, generateDMActionsForCommit(winningTxn3DomainMetadatas))
+
+      txn1.commit(engine, generateDMActionsForCommit(currentTxn1DomainMetadatas))
+
+      // If we reach here, there was no conflict
+      assert(!expectedConflict)
+
+      // Verify the final state includes merged domain metadata
+      val expectedMetadata =
+        (winningTxn2DomainMetadatas ++ winningTxn3DomainMetadatas ++ currentTxn1DomainMetadatas)
+          .groupBy(_.getDomain)
+          .mapValues(_.last)
+
+      assertDomainMetadata(table, engine, expectedMetadata)
+    } catch {
+      case e: KernelException =>
```

Comment:
does a thrown exception indicate that the test has passed? please use `val ex = intercept[KernelException] { ... code ... }`

Reply:
I've change this to use `val ex = intercept[KernelException] {...}`. It would be clearer this way.